scopes/harmony/application/deploy.task.ts

```diff
@@ -69,14 +80,22 @@ export class DeployTask implements BuildTask {
     const componentResults = buildTask?.componentsResults.find((res) =>
       res.component.id.isEqual(componentId, { ignoreVersion: true })
     );
-    /**
-     * @guysaar223
-     * @ram8
-     * TODO: we need to think how to pass private metadata between build pipes, maybe create shared context
-     * or create new deploy context on builder
-     */
+
     // @ts-ignore
-    const metadata = componentResults?._metadata.buildDeployContexts.find(
+    const metadata = componentResults?.metadata?.buildDeployContexts.find(
+      (ctx) => ctx.name === app.name && ctx.appType === app.applicationType
+    );
+
+    return metadata;
+  }
+
+  private getBuildMetadataFromCompObject(component: Component, app: Application) {
+    const builderData = component.state.aspects.get('teambit.pipelines/builder')?.data;
+    if (!builderData) return undefined;
+    const appData = builderData.aspectsData.find((aspectData) => aspectData.aspectId === ApplicationAspect.id);
+    if (!appData) return undefined;
```

Comment:
This should be moved to the builder aspect 
similar to the `getDataByAspect` that we already have there.
you shouldn't go here and find/get by providing the builder ID directly.

Reply:
actually it's `getDataByAspect` :) I didn't think it's the same thing. I'm adding docs above this method to explain it better.